server.py

```diff
@@ -186,28 +186,42 @@ def render(self, request):
         args = {k.decode(): [item for item in v] for k, v in request.args.items()}
 
         path = request.path.decode()
-        logger.info('path is {path}', path=path)
-        request.responseHeaders.addRawHeader(b"content-type", b"application/json")
-        if ('%s:%s' % (path, request.method.decode())) in allowable_methods:
-            ret = contacts.execute_route(path, data, args)
-            if 'error' in ret:
-                request.setResponseCode(ret.get('status', 400))
-                ret = ret['error']
-                logger.info('error return is {ret}', ret=ret)
-            else:
-                # if any values functions in ret, then run then asynchronously and return None here
-                # if they aren't then return ret
-
-                ret = resolve_all_functions(ret, request)
-                logger.info('legal return is {ret}', ret=ret)
-        else:
-            request.setResponseCode(402)
-            ret = {"error": "no such request"}
-            logger.info('return is {ret}', ret=ret)
-        if twserver.NOT_DONE_YET != ret:
-            return json.dumps(ret).encode()
+        method = request.method.decode()
+        path_method = '%s:%s' % (path, method)
+        logger.info('{method} {path}', method = method, path = path)
+        if method == "OPTIONS":
+            request.setResponseCode(204)
+            request.responseHeaders.addRawHeader(b"Allow", b"OPTIONS, GET, POST")
+            request.responseHeaders.addRawHeader(b"Access-Control-Allow-Methods", b"GET, POST, OPTIONS")
+            request.responseHeaders.addRawHeader(b"Connection", b"Keep-Alive")
+            request.responseHeaders.addRawHeader(b"Access-Control-Allow-Headers", b"Content-Type")
+            # The cross-origin headers were needed for some health provider portal work that isn't happening yet so commented out.
+            # request.responseHeaders.addRawHeader(b"Access-Control-Allow-Origin", b"*")
+            # request.responseHeaders.addRawHeader(b"Vary", b"Origin")
+            return b""
         else:
-            return ret
+            request.responseHeaders.addRawHeader(b"content-type", b"application/json")
+            request.responseHeaders.addRawHeader(b"access-control-allow-origin", b"*")
```

Comment:
ah, again, i think this should come from the config file.

Reply:
See above , but I'll comment that out anyway (unintentionally missed it)